File: traffic_violation/traffic_violation/report/violation_report/violation_report.js
// Copyright (c) 2025, Amr and contributors
// For license information, please see license.txt

frappe.query_reports["Violation Report"] = {
	"filters": [
		{
			"fieldname":"from_date",
			"label": __("From"),
			"fieldtype": "Date",
			"default": frappe.datetime.add_months(frappe.datetime.get_today(),-1),
			"reqd": 1,
			"width": "100px"
		},
		{
			"fieldname":"to_date",
			"label": __("To"),
			"fieldtype": "Date",
			"default": frappe.datetime.get_today(),
			"reqd": 1,
			"width": "100px"
		},
		{
			"fieldname":"violation_name",
			"label": __("Violation Name"),
			"fieldtype": "Link",
			"options": "Violation Type",
			"width": "100px"
		},


	]
};

File: traffic_violation/traffic_violation/report/violation_report/violation_report.py
# Copyright (c) 2025, Amr and contributors
# For license information, please see license.txt
import frappe
from frappe import _


def execute(filters=None):
	columns, data = [
				{
			"label": _("Number Plate"),
			"fieldname": "document_number",
			"fieldtype": "Link",
			"options": "Car Card",
			"width": 150,
		},
		{
			"label": _("Full Name"),
			"fieldname": "full_name",
			"fieldtype": "Data",
			"width": 120,
		},
		{
			"label": _("Violation Name"),
			"fieldname": "violation_name",
			"fieldtype": "Link",
			"options": "Violation Type",
			"width": 140,
		},
	], []
	return columns, data


